okio/jvm/src/main/java/okio/Pipe.kt

```diff
@@ -116,6 +180,36 @@ class Pipe(internal val maxBufferSize: Long) {
     override fun timeout(): Timeout = timeout
   }
 
+  /**
+   * Writes any buffered contents of this pipe to `sink`, then changes this pipe's sink to forward
+   * to `sink`. This pipe's source is closed and attempts to read it will throw
+   * [IllegalStateException].
+   *
+   * This method must not be called while concurrently accessing this pipe's sink. It is safe,
```

Comment:
This method must not be called while concurrently accessing this pipe's **sink**.
 -> This method must not be called while concurrently accessing this pipe's **source**.